test/helper.rb

```diff
@@ -135,18 +192,35 @@ def self.run(reporter, options = {}) # :nodoc:
     super
   end
 
+  def after_teardown
+    return if passed? || skipped?
+    out_io unless Puma.jruby? or RUBY_VERSION < '2.3'
+  end
+
   def full_name
     "#{self.class.name}##{name}"
   end
+
+  def full_file_name
+    cls_name = self.class.name.gsub(/([a-z])([A-Z])/, '\1_\2').downcase
+    "#{cls_name}__#{name}"
+  end
 end
 
 Minitest.after_run do
   # needed for TestCLI#test_control_clustered
   unless $debugging_hold
-    out = $debugging_info.strip
-    unless out.empty?
+    out = "#{$debugging_info.rstrip}\n".dup
```

Comment:
Should be extracted into its own method, like `assert_no_open_tmp_files`

Reply:
See above